### PR TITLE
feat(binlogsyncer): support set binlogsyncer maxreconnectattempts

### PR DESCRIPTION
**Pull request by shaohk**
## A Pull Request should be associated with an Issue.

> We wish to have discussions in Issues. A single issue may be targeted by multiple PRs.
> If you're offering a new feature or fixing anything, we'd like to know beforehand in Issues,
> and potentially we'll be able to point development in a particular direction.

Related issue: https://github.com/github/gh-ost/issues/0123456789

> Further notes in https://github.com/github/gh-ost/blob/master/.github/CONTRIBUTING.md
> Thank you! We are open to PRs, but please understand if for technical reasons we are unable to accept each and any PR

### Description

This PR [briefly explain what it does]

> In case this PR introduced Go code changes:

- [ ] contributed code is using same conventions as original code
- [ ] `script/cibuild` returns with no formatting errors, build errors or unit test errors.

### CODE REVIEW

File: go/cmd/gh-ost/main.go [hunk 1]
```diff
@@ -133,6 +133,7 @@ func main() {
 	flag.Int64Var(&migrationContext.HooksStatusIntervalSec, "hooks-status-interval", 60, "how many seconds to wait between calling onStatus hook")
 
 	flag.UintVar(&migrationContext.ReplicaServerId, "replica-server-id", 99999, "server id used by gh-ost process. Default: 99999")
+	flag.IntVar(&migrationContext.BinlogSyncerMaxReconnectAttempts, "binlogsyncer-max-reconnect-attempts", 0, "when master node fails, the maximum number of binlog synchronization attempts to reconnect. 0 is unlimited")
```
Comment (timvaillancourt) [inline go/cmd/gh-ost/main.go:136]: Can you please document this flag in `doc/command-line-flags.md`? 🙇

**Review by timvaillancourt — CHANGES_REQUESTED** [review]
@shaohk thanks for the PR! I added one comment requesting markdown documentation 🙇

File: go/cmd/gh-ost/main.go (hunk 1 above)
Comment (shaohk) [reply go/cmd/gh-ost/main.go:136]: OK

File: doc/command-line-flags.md [hunk 2]
```diff
@@ -61,6 +61,9 @@ It is not reliable to parse the `ALTER` statement to determine if it is instant
 
 `gh-ost` will automatically fallback to the normal DDL process if the attempt to use instant DDL is unsuccessful.
 
+### binlogsyncer-max-reconnect-attempts
+`--binlogsyncer-max-reconnect-attempts=0`, the maximum number of attempts to re-establish a broken connection for sync binlog. `0` or `negative number` means infinite retry, default `0`.
```
Comment (timvaillancourt) [inline doc/command-line-flags.md:65]:
```suggestion
`--binlogsyncer-max-reconnect-attempts=0`, the maximum number of attempts to re-establish a broken inspector connection for sync binlog. `0` or `negative number` means infinite retry, default `0`.
```

**Review by timvaillancourt — CHANGES_REQUESTED** [review]
@shaohk added 2 x final nits. LGTM otherwise

**Pull request closed without merging**